src/Symfony/Component/PropertyInfo/WriteMutator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyInfo;
+
+/**
+ * Writes mutator tell how to write to a property.
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+final class WriteMutator
+{
+    public const TYPE_METHOD = 1;
+    public const TYPE_PROPERTY = 2;
+    public const TYPE_ARRAY_DIMENSION = 3;
+    public const TYPE_ADDER_REMOVER_METHOD = 4;
+    public const TYPE_CONSTRUCTOR = 5;
+
+    private $type;
+    private $name;
+    private $adderName;
+    private $removerName;
+    private $private;
+    private $constructorParameter;
+    private $static;
+
+    public function __construct(int $type, ?string $name, ?string $adderName, ?string $removerName, bool $private = false, bool $static = false, \ReflectionParameter $constructorParameter = null)
```

Comment:
this is a bit fragile. we could make the constructor private and have static create<Type> methods with all required arguments for that type.

```php
public static createMethod(string $methodName); // could this be private?
public static createProperty(string $propertyName, bool $private, bool $private); 
public static createArrayDimension(string $key);
public static createAdderRemoverMethod(string $adderName, string $removerName);
public static createConstructorArgument($constructorParameter);

Reply:
Not sure if applicable: Need to test this with how the mutator / accessor is created in ReflectionExtractor.

Reply:
WriteMutator / ReadAccessor now use static constructor for better UX